src/oop/special_class_method/medium/q10/MainTest.java

```diff
@@ -0,0 +1,275 @@
+            int size = (int) tests[i][0];
+            String colour = (String) tests[i][1];
+            Object[][] arguments = {
+                    {size, int.class},
+                    {colour, String.class}
+            };
+            Class<?>[] classes = {int.class, String.class};
+            Object testInstance = testClass.createInstance(arguments);
+            _assertEquals(size, testClass.getFieldValue(testInstance, attributeName1),
+                    String.format(modifiedClassMessage, className));
+            _assertEquals(colour, testClass.getFieldValue(testInstance, attributeName2),
+                    String.format(modifiedClassMessage, className));
+            assertTrue(testClass.hasModifier(classes, "public"),
+                    String.format(modifiedClassMessage, className));
+        }
+    }
+
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(3, "Red"),
+                Arguments.of(10, "Black"),
+                Arguments.of(12, "Blue"),
+                Arguments.of(43, "White"),
+                Arguments.of(7, "Yellow")
+        );
+    }
+
+    private static Stream<Arguments> updateIntegerInputProvider() {
+        return Stream.of(
+                Arguments.of(3, "Red", 5),
+                Arguments.of(10, "Black", 12),
+                Arguments.of(12, "Blue", 10),
+                Arguments.of(43, "White", 37),
+                Arguments.of(7, "Yellow", 4)
+
+        );
+    }
+
+    private static Stream<Arguments> updateStringInputProvider() {
+        return Stream.of(
+                Arguments.of(3, "Red", "White"),
+                Arguments.of(10, "Black", "Green"),
+                Arguments.of(12, "Blue", "Red"),
+                Arguments.of(43, "White", "Gray"),
+                Arguments.of(7, "Yellow", "Orange")
+
+        );
+    }
+
+    @Test
+    public void getSizeIsDefinedCorrectly() {
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(getAttributeMethodName1, null), String.format(incorrectMethodDefinition, getAttributeMethodName1, className));
+        assertTrue(testClass.hasModifier(getAttributeMethodName1, null, "public"), String.format(incorrectModifierMessage, getAttributeMethodName1, className));
+        assertTrue(testClass.hasReturnType(getAttributeMethodName1, null, int.class), String.format(incorrectReturnType, getAttributeMethodName1, className));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void correctGetSizeMethod(int value1, String value2) throws Throwable {
```

Comment:
For tests where only one value is checked switch to not using varying for the other value (ie. in this case for the arguments replace value2 in arguments with a default String value)